### PR TITLE
Allow gzip encoding for request services

### PR DESCRIPTION
**Pull request by msujew**
#### What it does

Closes https://github.com/eclipse-theia/theia/issues/11233

Allow `gzip` encoding by default for all requests coming from the `NodeRequestService`. Sends the respective `Accept-Encoding` headers and respects the `content-encoding` response header.

#### How to test

1. Confirm (debug) that the headers are present on request and response headers.
2. `download:plugins` cli command should still work as expected (maybe a bit faster)

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by msujew** [comment]
cc @ccorn Can you try whether this alleviates the performance regressions you were experiencing?

**Comment by ccorn** [comment]
If `zstd` were available as well, that [could reduce the huge transfer sizes](https://github.com/eclipse/openvsx/issues/379#issuecomment-1125263546) of some responses by a factor of 80 in comparison to `gzip`. Plus, it's faster. So, if the libs provide that feature already, using it would be beneficial. If (and only if) that is a low-hanging fruit, I'd recommend plucking it.

**Comment by ccorn** [comment]
Looking at the code of this PR, it seems that it re-implements request-response analysis. But the OVSX client had compression supported previously. Which leads me to the question: Isn't there a simple pass-through for the no-proxy case available? That would be more robust, not just with respect to compression headers but to all other conceivable specifics, I think.

**Comment by msujew** [comment]
@ccorn Thanks for testing it!

Regarding the `zstd` change: I would be all-in for it, if Node would provide a built-in lib for that (like they do for `gzip`). But as it stands I would wait until ovsx supports it.

In general regarding using a more sophisticated http-library: We could do that, but lose some of the fine-grained nature of using the plain http package. I've seen that `node-fetch` now actually supports proxy agents, so it might worthwhile to investigate. For now, I would stick with the low-level approach.

File: dev-packages/request/src/node-request-service.ts [hunk 1]
```diff
@@ -112,11 +119,17 @@ export class NodeRequestService implements RequestService {
                 } else {
                     const chunks: Uint8Array[] = [];
 
-                    res.on('data', chunk => {
+                    let stream: Stream = res;
+
+                    if (res.headers['content-encoding'] === 'gzip') {
+                        stream = stream.pipe(createGunzip());
+                    }
```
Comment (colin-grant-work) [inline dev-packages/request/src/node-request-service.ts:126]: I may be a bit too big a fan of ternaries, but here it seems reasonable, unless we anticipate other cases to handle?
```suggestion
                    const stream = res.headers['content-encoding'] === 'gzip' ? res.pipe(createGunzip()) : res;
```

**Review by colin-grant-work — APPROVED** [review]
Code looks reasonable and no problems on my end. It did speed up the plugin download noticeably, even on a pretty fast network.

**Pull request closed without merging**